pkg/engine/lifecycletest/protect_test.go

```diff
@@ -173,3 +175,156 @@ func TestProtectInheritance(t *testing.T) {
 	assert.Equal(t, snap.Resources[0].URN, snap.Resources[2].Parent)
 	assert.Equal(t, snap.Resources[0].URN, snap.Resources[3].Parent)
 }
+
+// TestProtectedDeleteChainsWithDuplicateDeletedResources tests that delete chains that error out due to protected
+// resources correctly abandon resources left in the chain, even in the presence of other resources with the same URN
+// (such as can occur when a replace is interrupted leaving both old and new copies of a resource in the snapshot).
+func TestProtectedDeleteChainsWithDuplicateDeletedResources(t *testing.T) {
+	t.Parallel()
+
+	// Arrange.
+
+	p := &lt.TestPlan{
+		Project: "test-project",
+		Stack:   "test-stack",
+	}
+	project := p.GetProject()
+
+	// Set up the initial snapshot. We want to end up with the following starting state:
+	//
+	// * A
+	// * B, which depends on A through a DeletedWith dependency, and has Protect: true
```

Comment:
Can we not use DeletedWith. I think you've found another bug here with destroy deleting B when it knows its going to delete A as well and so should skip the B delete.